clang/lib/CodeGen/CGDebugInfo.cpp

```diff
@@ -3424,6 +3445,26 @@ llvm::DIMacroFile *CGDebugInfo::CreateTempMacroFile(llvm::DIMacroFile *Parent,
   return DBuilder.createTempMacroFile(Parent, Line, FName);
 }
 
+llvm::DILocation *CGDebugInfo::CreateTrapFailureMessageFor(
```

Comment:
There's currently only one caller with a fixed parameter for `Prefix` - perhaps that could be hardcoded in the implementation instead?

Reply:
@dwblaikie We currently use this function for `-fbounds-safety` internally. We are currently in the process of upstreaming this which is why the function should **not** hardcode the `Prefix`.

Reply:
@dwblaikie what do you think? `-fbounds-safety` is currently being upstreamed and there is going to be another caller of the function in the not-too-distant future.

Reply:
@dwblaikie any comments?

Reply:
I think I agree with https://github.com/llvm/llvm-project/pull/79230/files#r1529429173 then - the prefix should be a parameter and I don't think it needs to be a #defined constant (CLANG_VERBOSE_TRAP_PREFIX) - and can instead be written as a literal in the caller in CGBuiltin.cpp?

Reply:
OK, I'll make it a parameter then.

I added macro `CLANG_VERBOSE_TRAP_PREFIX` to `ModuleBuilder.h` so that lldb wouldn't have to define it too. See https://github.com/llvm/llvm-project/pull/80368#discussion_r1481781470.

Reply:
@dwblaikie 's comment:
https://github.com/llvm/llvm-project/pull/79230#discussion_r1575391593

So the name of the subprogram would be something like `__builtin_trap: identifier123 : Argument_must_not_be_null`?

@Michael137 @delcypher is there a reason we cannot use the same prefix (`__builtin_trap`, for example)? Does lldb need clang to use different prefixes for `-fbounds-safety` and `builtin_verbose_trap`?

Reply:
I've updated the patch to pass a category string to the builtin. The artificial function name has the format `<prefix>:<category>:<message>` where `<prefix>` is always `__builtin_verbose_trap`.

We should probably use something other than `:` for the separator as users might want to use it in the category or message string (e.g., `boost::some_library_name`).

Reply:
@dwblaikie any other comments?

Reply:
Eh, I'm not too fussed about the separator for the prefix (we control that) and category (will the category be user-visible? Like a warning group? Or is that only still an implementation detail/contract between DWARF producer and DWARF consumer? I thought it was more the latter/implementation detail? In which case we can say what characters are valid there, and I'd keep it pretty simple, like the warning group names - all lower, dash or underscore separated seems fine)